### ISSUE TITLE
Error: Incorrect attribute value type deploying on AWS with Terraform

### ISSUE THREAD

**Issue by edernucci** [kubermatic/kubeone#463]
**What happened**:
Error running `terraform plan`.

**What is the expected behavior**:
`terraform plan` should execute without errors.

**How to reproduce the issue**:
```
$ terraform plan

Error: Incorrect attribute value type

  on main.tf line 180, in resource "aws_lb" "control_plane":
 180:   subnets            = ["${local.all_subnets}"]

Inappropriate value for attribute "subnets": element 0: string required.
```

**Anything else we need to know?**
Using files from https://github.com/kubermatic/kubeone/tree/master/examples/terraform/aws

```
$ terraform version
Terraform v0.12.0
+ provider.aws v2.11.0
```
```
$ cat terraform.tfvars 
cluster_name = "demo"
aws_region = "sa-east-1"
```

**Information about the environment**:
KubeOne version (`kubeone version`): 
```
$ kubeone version
{
  "kubeone": {
    "major": "0",
    "minor": "6",
    "gitVersion": "0.6.2",
    "gitCommit": "7b86c3bf654366a64da608c7b710360dd5f5d777",
    "gitTreeState": "",
    "buildDate": "2019-05-13T16:49:50Z",
    "goVersion": "go1.12.5",
    "compiler": "gc",
    "platform": "linux/amd64"
  },
  "machine_controller": {
    "major": "1",
    "minor": "1",
    "gitVersion": "v1.1.5",
    "gitCommit": "",
    "gitTreeState": "",
    "buildDate": "",
    "goVersion": "",
    "compiler": "",
    "platform": "linux/amd64"
  }
}
```

Operating system: 
```
$ uname -a
Linux artorias 4.9.0-9-amd64 #1 SMP Debian 4.9.168-1+deb9u2 (2019-05-13) x86_64 GNU/Linux
```
Provider you're deploying cluster on:
AWS
Operating system you're deploying on:
Linux

**Comment by kron4eg** [comment]
Thanks for the report, terraform 0.12 just got release and it seems like out _example_ terraform configs are only 0.11 compatible (despite `terraform 0.12checklist` claimed otherwise). Most likely we going to copy existing configs and make copies 0.12 compatible, but that'll take some time.

**Comment by edernucci** [comment]
@kron4eg 

This was my first experience with Terraform, so I thought I was doing something wrong.

In this case if I use version 0.11 everything should work smoothly?

**Comment by kron4eg** [comment]
yeah, we use 0.11 terraform in out e2e tests and it works quite well for us.

**Comment by edernucci** [comment]
@kron4eg 

Just for clarification, I downgraded Terraform, cleaned the directory and started again.

Everything worked as expected with version 0.11.14.

Thanks for your support.

**Comment by edernucci** [comment]
**But** (in software we always have a "but") I'm stuck with this:

```
$ terraform refresh
aws_key_pair.deployer: Refreshing state... (ID: demo-deployer-key)
aws_default_vpc.default: Refreshing state... (ID: vpc-026a58a168c5e8b1b)
aws_iam_role.role: Refreshing state... (ID: demo-host)
data.aws_availability_zones.available: Refreshing state...
data.aws_ami.ubuntu: Refreshing state...
aws_iam_role_policy.policy: Refreshing state... (ID: demo-host:demo-host)
aws_iam_instance_profile.profile: Refreshing state... (ID: demo-host)
aws_security_group.control_plane: Refreshing state... (ID: sg-0ba14f01b6872c98a)
data.aws_subnet.az_a: Refreshing state...
aws_lb_target_group.control_plane_api: Refreshing state... (ID: arn:aws:elasticloadbalancing:sa-east-1:...:targetgroup/demo-api/12896cf7f8db0d64)
aws_security_group.common: Refreshing state... (ID: sg-0c41d155b2ed9a252)
data.aws_subnet_ids.default: Refreshing state...
data.aws_subnet.az_c: Refreshing state...
data.aws_subnet.az_b: Refreshing state...

Error: Error refreshing state: 1 error occurred:
	* data.aws_subnet.az_b: 1 error occurred:
	* data.aws_subnet.az_b: data.aws_subnet.az_b: no matching subnet found
```
I already deleted every resource and tried with many options, but it can't run.

Any ideas?

**Comment by eqrx** [comment]
@edernucci Terraform 0.12 support have just been merged and will be available with the 0.9 release (Should come around the end of this week)
